packages/handler-next/src/createNextSynthqlHandler.ts

```diff
@@ -46,33 +67,52 @@ async function executeSynthqlRequest<DB>(
     req: NextSynthqlHandlerRequest,
 ): Promise<NextResponse> {
     // First try to parse the request body as JSON
-    const { query, returnLastOnly } = await tryParseRequest(req);
+    const { body, headers } = await tryParseRequest(req);
 
     // We don't do this yet, but eventually we'll want to validate the request
     // const validatedQuery = await tryValidateSynthqlQuery(query);
 
     // Execute the query, but just to get the initial generator
-    const resultGenerator = await tryExecuteQuery<DB>(
-        queryEngine,
-        query,
-        returnLastOnly,
-    );
+    const resultGenerator = isRegisteredQueryRequest(body)
+        ? await tryExecuteRegisteredQuery<DB>(
+              queryEngine,
+              { queryId: body.queryId, params: body.params },
+              headers.returnLastOnly,
+          )
+        : isRegularQueryRequest(body)
```

Comment:
All this execution logic should be moved to a single tryExecuteQuery (pseudocode):

```
function tryExecuteQuery(q) {
    if isRegisteredQuery(q) { queryEngine.executeRegisteredQuery(...) }
    else {
        queryEngine.executeQuery(...)
    }
}
```